tests/v2prov/tests/machineprovisioning/machine_test.go

```diff
@@ -649,3 +651,171 @@ func Test_Provisioning_MP_DrainNoDelete(t *testing.T) {
 	err = cluster.EnsureMinimalConflictsWithThreshold(clients, c, cluster.SaneConflictMessageThreshold)
 	assert.NoError(t, err)
 }
+
+func Test_Provisioning_Single_Node_All_Roles_Drain(t *testing.T) {
+	t.Parallel()
+
+	clients, err := clients.New()
+	require.NoError(t, err)
+	defer clients.Close()
+
+	ctx := clients.Ctx
+
+	drainOptions := rkev1.DrainOptions{
+		Enabled:                         true,
+		DeleteEmptyDirData:              true,
+		DisableEviction:                 false,
+		GracePeriod:                     -1,
+		Force:                           false,
+		IgnoreDaemonSets:                ptr.To(true),
+		SkipWaitForDeleteTimeoutSeconds: 0,
+		Timeout:                         120,
+	}
+
+	// Single-node (cp+etcd+worker) with drain-on-config enabled
+	base := &provisioningv1api.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-single-node-drain",
+		},
+		Spec: provisioningv1api.ClusterSpec{
+			KubernetesVersion: defaults.SomeK8sVersion,
+			RKEConfig: &provisioningv1api.RKEConfig{
+				ClusterConfiguration: rkev1.ClusterConfiguration{
+					UpgradeStrategy: rkev1.ClusterUpgradeStrategy{
+						ControlPlaneDrainOptions: drainOptions,
+						ControlPlaneConcurrency:  "1",
+						WorkerDrainOptions:       drainOptions,
+						WorkerConcurrency:        "1",
+					},
+				},
+				MachinePools: []provisioningv1api.RKEMachinePool{{
+					EtcdRole:         true,
+					ControlPlaneRole: true,
+					WorkerRole:       true,
+					Quantity:         &defaults.One,
+				}},
+			},
+		},
+	}
+
+	c, err := cluster.New(clients, base)
+	require.NoError(t, err)
+
+	c, err = cluster.WaitForCreate(clients, c)
+	require.NoError(t, err)
+
+	// Validate that exactly one Machine and capture its template hash
+	machines, err := cluster.Machines(clients, c)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(machines.Items), "expected exactly one machine initially")
+
+	old := machines.Items[0]
+	oldHash := old.Labels["machine-template-hash"]
+	require.NotEmpty(t, oldHash, "old machine missing template-hash label")
+
+	// Create a fresh machine config that can be mutated.
+	newCfgRef, err := nodeconfig.NewPodConfig(clients, c.Namespace)
+	require.NoError(t, err)
+
+	gvrPodCfg := schema.GroupVersionResource{
+		Group: "rke-machine-config.cattle.io", Version: "v1", Resource: "podconfigs",
+	}
+	pc, err := clients.Dynamic.Resource(gvrPodCfg).Namespace(c.Namespace).Get(ctx, newCfgRef.Name, metav1.GetOptions{})
```

Comment:
```suggestion
	podconfig, err := clients.Dynamic.Resource(gvrPodCfg).Namespace(c.Namespace).Get(ctx, newCfgRef.Name, metav1.GetOptions{})
```

alternatively `podConfig`, I don't know where we stand on it being one word or not lol. I'll leave it to your discretion.